src/instruction/binary_op.rs

```diff
@@ -17,19 +17,12 @@ pub struct BinaryOp {
     lhs: Box<dyn Instruction>,
     rhs: Box<dyn Instruction>,
     op: Operator,
-
-    value: Option<Box<dyn Instruction>>,
```

Comment:
Is this for memoization purposes ? Or is there another reason to store the result ?

Reply:
I think this is a relic of when the binary ops were first implemented. I deleted it because we're not using it anywhere